Expensify/App
# Set password

### Details
Create a new UX around setting a password during account signup. 
Giving feedback to the user before submission on:
password complexity requirements
passwords matching


### Fixed Issues
Fixes https://github.com/Expensify/Expensify.cash/issues/1517


### QA Steps
To get to this page:
trigger password reset
make sure you're logged out
navigate to here: http://localhost:8080/setpassword/[USERID]/[RESETCODE]
make sure you have credentials.login key set to your email address in local storage

Then follow the video:
https://user-images.githubusercontent.com/47184118/120825774-a23f1c80-c551-11eb-9614-a5cf2f00414e.mp4

### Testing the form

Try to break the form by different combinations e.g. put in an invalid password, then replicate that in the next box. Then make the first password valid and see if the red hint turns green. And see if the passwords match error disappears. 

- You shouldn't be able to submit the form if the passwords don't match or the password doesn't meet the complexity requirements.
- The hint should only turn red if an invalid password is entered and the focus leaves the input. After this condition has been met it should give real time feedback if this requirement is met or not.
- The hint should turn green when a valid password is entered.
- The passwords should match error should only show if the confirm box input loses focus while the passwords don't match.  After this condition has been met it should give real time feedback if this requirement is met or not.


### Testing  submission:
When pressing the Set Password button:
If you use an invalid password or they don't match the set password button shouldn't do anything.
If you use an expired code the Validated code is likely expired error should shot.
If you used a password that meets the complexity then you should be logged into the app and your password will have been set/changed

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
